servers/jax-rs-tests/src/main/java/org/projectnessie/jaxrs/tests/AbstractRestMergeTransplant.java

```diff
@@ -364,6 +365,126 @@ private void conflictExceptionReturnedAsMergeResult(
+        ImmutableList.of(
+            "test-commit-1\n---------------------------------------------\ntest-commit-2"));
+  }
+
+  @Test
+  @NessieApiVersions(versions = NessieApiVersion.V1) // V2 does not allow squashed transplants
+  public void transplantMessageSquashed() throws BaseNessieClientServerException {
+    testMergeTransplantMessage(
+        (target, source, committed1, committed2, returnConflictAsResult) ->
+            getApi()
+                .transplantCommitsIntoBranch()
+                .message("test-message-override-123")
+                .branch(target)
+                .fromRefName(source.getName())
+                .keepIndividualCommits(false)
+                .hashesToTransplant(
+                    ImmutableList.of(
+                        Objects.requireNonNull(committed1.getHash()),
+                        Objects.requireNonNull(committed2.getHash())))
+                .returnConflictAsResult(returnConflictAsResult)
+                .transplant(),
+        ImmutableList.of("test-message-override-123"));
+  }
+
+  @Test
+  @NessieApiVersions(versions = NessieApiVersion.V1) // V2 does not allow squashed transplants
+  public void transplantMessageSingle() throws BaseNessieClientServerException {
+    testMergeTransplantMessage(
+        (target, source, committed1, committed2, returnConflictAsResult) ->
+            getApi()
+                .transplantCommitsIntoBranch()
+                .message("test-message-override-123")
+                .branch(target)
+                .fromRefName(source.getName())
+                .hashesToTransplant(ImmutableList.of(Objects.requireNonNull(committed1.getHash())))
+                .returnConflictAsResult(returnConflictAsResult)
+                .transplant(),
+        ImmutableList.of("test-message-override-123"));
+  }
+
+  @Test
+  public void transplantMessageOverrideMultiple() throws BaseNessieClientServerException {
+    testMergeTransplantMessage(
+        (target, source, committed1, committed2, returnConflictAsResult) ->
+            getApi()
+                .transplantCommitsIntoBranch()
+                .message("ignored-message-override")
+                .keepIndividualCommits(true)
+                .branch(target)
+                .fromRefName(source.getName())
+                .hashesToTransplant(
+                    ImmutableList.of(
+                        Objects.requireNonNull(committed1.getHash()),
+                        Objects.requireNonNull(committed2.getHash())))
+                .returnConflictAsResult(returnConflictAsResult)
+                .transplant(),
+        // Note: the expected messages are given in the commit log order (newest to oldest)
+        ImmutableList.of("test-commit-2", "test-commit-1"));
+  }
+
+  private void testMergeTransplantMessage(
+      MergeTransplantActor actor, Collection<String> expectedMessages)
+      throws BaseNessieClientServerException {
+    Branch target = createBranch("merge-transplant-msg-target");
```

Comment:
Please please add a common ancestor commit for this test

Reply:
Will do a fix here. Also: https://github.com/projectnessie/nessie/pull/5696

Reply:
fixed

Reply:
thanks!